### ISSUE TITLE
No Land-Atmosphere Carbon (CO2 only?) Fluxes in Coupled Model?

### ISSUE THREAD

**Issue by rgknox** [NGEET/fates#163]
### Summary of Issue:

As far as I can tell, FATES does not pass the requisite information to the host land model to run coupled with an atmosphere.  I think the only deficiencies are carbon from co2 and fire emissions.

I believe that carbon fluxes (as well as all other grid-scale fluxes) are prepared in the call to lnd2atm() during clm_drv().  Prior to lnd2atm(), a grid-scale vector called net_carbon_exchange_grc is filled, and I believe this is the key land-atmosphere (non-fire) carbon flux variable.  I don't think fates is sending information to the host model (nep, fire carbon fluxes) that will fill this variable.  Can anyone see otherwise?

I think coupled land-atmosphere simulations with FATES is pre-mature right now.  But I think we should all be well aware that the fluxes are not even hooked up.

### Expected behavior:

net_carbon_exchange_grc(g) should be populated with nep fluxes?
fireemis_inst should also be populated with fire fluxes.

**Comment by ckoven** [comment]
The thing missing right now is to add in the fire carbon flux to the total land-atmosphere flux term.  that variable is:
https://github.com/NGEET/fates/blob/master/main/EDTypesMod.F90#L668

**Comment by rosiealice** [comment]
Is this still a thing? The line in the link seems not to work anymore. I guess it moved somewhere else?

**Comment by rosiealice** [comment]
@rgknox is this still a thing??

**Comment by rgknox** [comment]
Yes, this is definitely still a thing. On both counts.  Fates only informs CLM/ELM of fragmented litter fluxes. It does not inform the hosts regarding net differences in atmospheric co2 exchange (via gpp and AR), or carbon released to the atm through burning, since these exchanges are not necessary for land-only runs not coupled with the atmosphere.

**Comment by rosiealice** [comment]
I am working on this here:
https://github.com/NorESMhub/CTSM/compare/noresm...rosiealice:ctsm:nbp_output?expand=1

and  here
https://github.com/NorESMhub/fates/compare/noresm...rosiealice:fates:nbp_outputs?expand=1

The code is running and generating NBP outputs. It does not yet work with the product pools (although the current spinup is not using landuse) nor does it have a balance check...
